platform/innovium/sonic-platform-modules-cameo/esc601-32q/modules/x86-64-cameo-esc601-32q.h

```diff
@@ -0,0 +1,647 @@
+
+/* end of i2c_client Declaration */
+
+/* Function Declaration */
+/* i2c-0 */
+#if 0
+static ssize_t tlv_status_get(struct device *dev, struct device_attribute *da, char *buf);
+#endif
+static ssize_t psu_status_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t psu_Pin_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t psu_Pout_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t psu_Vin_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t psu_Vout_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t psu_Iin_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t psu_Iout_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t usb_power_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t usb_power_set(struct device *dev, struct device_attribute *da, const char *buf, size_t count);
+#ifdef LED_CTRL_WANTED
+static ssize_t led_ctrl_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t led_ctrl_set(struct device *dev, struct device_attribute *da, const char *buf, size_t count);
+#endif
+static ssize_t led_loc_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t led_loc_set(struct device *dev, struct device_attribute *da, const char *buf, size_t count);
+static ssize_t led_alarm_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t led_alarm_set(struct device *dev, struct device_attribute *da, const char *buf, size_t count);
+static ssize_t reset_mac_set(struct device *dev, struct device_attribute *da, const char *buf, size_t count);
+static ssize_t themal_status_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t themal_temp_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t themal_mask_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t themal_mask_set(struct device *dev, struct device_attribute *da, const char *buf, size_t count);
+static ssize_t int_status_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t low_power_all_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t low_power_all_set(struct device *dev, struct device_attribute *da, const char *buf, size_t count);
+static ssize_t low_power_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t low_power_set(struct device *dev, struct device_attribute *da, const char *buf, size_t count);
+static ssize_t qsfp_reset_set(struct device *dev, struct device_attribute *da, const char *buf, size_t count);
+static ssize_t qsfp_status_get(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t qsfp_temp_get(struct device *dev, struct device_attribute *da, char *buf);
```

Comment:
use optoe driver.

Reply:
> use optoe driver.

How to use optoe driver? Can you give an example? I have read about optoe driver source code,
but seem not fit our hardware arch.

Reply:
> as comments

Already updated to use optoe driver for accessing qsfp module.